test/persistence-hooks.suite.js

```diff
@@ -3240,13 +3249,17 @@ module.exports = function(dataSource, should, connectorCapabilities) {
           { id: existingInstance.id, name: 'updated name' },
           function(err, instance) {
             if (err) return done(err);
-            ctxRecorder.records.should.eql(aCtxForModel(TestModel, {
+            var expectedContext = aCtxForModel(TestModel, {
               data: {
                 id: existingInstance.id,
                 name: 'updated name',
               },
-              isNewInstance: false,
-            }));
+            });
+
+            if (dataSource.connector.upsertWithWhere) {
+              expectedContext.isNewInstance = false;
```

Comment:
Fair enough. Please capture this information in a comment in the source code to prevent confusion of future readers.